### ISSUE TITLE
RGB Dataset 224* 224 dimensions

### ISSUE THREAD

**Issue by redhat12345** [XifengGuo/CapsNet-Keras#42]
How can we use ur code in other RGB dataset?
Suppose the structure of dataset is like that. it contains some sub-folder. Each sub-folder represents one class.

Class A:
  0001.jpg    1
  0002.jpg    1
Class B:
  0001.jpg    2
  0002.jpg    2

**Comment by redhat12345** [comment]
Thank you for your answer. I have changed but got an error. I want to train my own data to use your code and generate images as well. can you help me please how can I do that?

   generator = train_datagen.flow_from_directory(directory=/media/user1/DATA/capsnet-keras/train_data_folder, target_size=(224, 224), batch_size=batch_size)

**Comment by XifengGuo** [comment]
@redhat12345 You should first learn how to use ImageDataGenerator from keras.io or Google, and then debug your errors.

**Comment by Ahmedest61** [comment]
@XifengGuo  In **load_mnist()**, for 3 channel image and different dimensions let say (128x128) do we need to reshape the features? Since currently if we do it reshape to let say (x, 128, 128, 1) from (y, 128, 128, 3) ? What is the purpose of reshaping ?

**Comment by Ahmedest61** [comment]
Also in **load_mnist()**, you have used **to_categorical()** which consider digits like 0 to 9 and change the shape to 10 but in our case if we have let say 43 labels/classes, it change it to 44 which is **wrong.** When it calls **_standardize_input_data()** in **_standardize_user_data()**, it gives an error of data shape and shape not matched i.e 44 is not matched with 43 (which is the actual number of classes). Any solution for this?
